File: 1319-UniqueNumberOfOccurrences/1319-UniqueNumberOfOccurrences.java
// Last updated: 3/26/2026, 1:50:04 AM
class Solution {
    public boolean uniqueOccurrences(int[] arr) {
      int[] test = new int[2001];
        int count =0;
        for(int i=0;i<arr.length;i++){
            int index;
            if(arr[i]<0){
              index = 1000-arr[i];  
            }
            else{
               index = arr[i]; 
            }
            test[index]++;
                if(test[index]==1)
                 count++;
        }
         ArrayList<Integer> list = new ArrayList<Integer>(); 
  
        for (int i = 0; i < test.length; i++) 
            list.add(new Integer(test[i])); 
         Set<Integer> set = new HashSet<Integer>(list);
  
        System.out.println(set);
        return (set.size()-1)==count;
    }
}

File: 1319-UniqueNumberOfOccurrences/1319-UniqueNumberOfOccurrences.cs
// Last updated: 3/26/2026, 1:50:01 AM
public class Solution {
    public bool UniqueOccurrences(int[] arr) {
        Dictionary<int,int> occ = new Dictionary<int, int>();
        
        for(int i=0; i< arr.Length; i++)
        {
            if (occ.ContainsKey(arr[i])) { occ[arr[i]]++; } 
            else { occ[arr[i]] = 1; }            
        }
        
        HashSet<int> unique = new HashSet<int>();
        foreach(int key in occ.Keys)
        {
            if (!unique.Add(occ[key])) { return false; }
        }
         
        return true;
    }
}